### DISCUSSION TITLE
eglot+gopls is ... off by one expression?

### DISCUSSION THREAD

**Discussion by Cerebus** [joaotavora/eglot#1100] [Q&A] [answered]
I really don't know what's happening here, and I'll try to generate a minimal example, but by screenshots:

- With a previous expression of the same kind, autocomplete suggestions are correct:
![with-previous-expression](https://user-images.githubusercontent.com/49740/200853387-8da5e744-8556-4f9c-a589-6a9c45db0c91.png)

- Without a previous expression...
![no-previous-expression](https://user-images.githubusercontent.com/49740/200853508-c3babd95-1ad7-4829-96b9-c2fc21993035.png)

I tried mucking about with `eglot--pos-to-lsp-position` but I don't really know if that's the problem.  I'll also generate a minimal example when I have a chance, but mainly looking for feedback on whether this is an eglot or gopls problem.

**Comment by Cerebus** [comment]
Absolutely minimal example:
```
~/Documents/projects/newgo main
> ls -la
total 16
drwxr-xr-x   5 tmiller  staff   160 Nov  9 09:05 .
drwxr-xr-x@ 92 tmiller  staff  2944 Nov  9 09:05 ..
drwxr-xr-x  10 tmiller  staff   320 Nov  9 09:05 .git
-rw-r--r--   1 tmiller  staff    33 Nov  9 09:06 foo.go
-rw-r--r--   1 tmiller  staff    28 Nov  9 09:04 go.mod
```

`foo.go` contains:
```
package main

import (
	"fmt"
)
```

Still get the same weird errors:

![Screen Shot 2022-11-09 at 9 15 50 AM](https://user-images.githubusercontent.com/49740/200868452-9cf4b2b8-c504-4b40-90f4-d0052641160e.png)

**Comment by Cerebus** [comment]
[eglot-log.txt](https://github.com/joaotavora/eglot/files/9972403/eglot-log.txt)

**Comment by Cerebus** [comment]
There's clearly a positioning issue between the buffer and the lsp.  When I can get an autocompletion actually suggested, it inserts at the wrong point.

https://user-images.githubusercontent.com/49740/200872123-2fd30e0f-8eca-473f-8b6a-c96f872424bd.mov

**Comment by joaotavora** [comment, marked as answer]
This was later found to be caused by the user's Emacs configuration which has a lot of extensions.

**Reply by Cerebus** [reply]
Turned out to be doom-modeline for reasons that remain totally unclear.
